OpenSim/Region/Framework/Scenes/ScenePresence.cs

```diff
@@ -564,13 +577,52 @@ public override Vector3 AbsolutePosition
                         if (IsBot && !Util.IsValidRegionXY(ppos))
                         {
                             Util.ForceValidRegionXY(ref ppos);
+                            m_physicsActor.Velocity = Vector3.Zero;
                             posForced = true;
                         }
+
+                        ILandObject parcel = Scene.LandChannel.GetLandObject(ppos.X, ppos.Y);
+                        if (parcel != null)
+                        {
+                            ParcelPropertiesStatus reason;
+                            if ((ppos.Z < Scene.LandChannel.GetBanHeight()) && (parcel.DenyParcelAccess(this.UUID, out reason)))
```

Comment:
Since this block is called by physics updating position info 64x/sec, have we checked to make sure parcel.DenyParcelAccess isn't going to the database etc?